internal/distro/fedora/pipelines.go

```diff
@@ -259,213 +257,117 @@ func iotContainerPipelines(t *imageType, customizations *blueprint.Customization
-	}
+	pl.Firewall = c.GetFirewall()
 
-	if pwQuality := imageConfig.PwQuality; pwQuality != nil {
-		p.AddStage(osbuild.NewPwqualityConfStage(pwQuality))
+	language, keyboard := c.GetPrimaryLocale()
+	if language != nil {
+		pl.Language = *language
+	} else {
+		pl.Language = imageConfig.Locale
 	}
-
-	if waConfig := imageConfig.WAAgentConfig; waConfig != nil {
-		p.AddStage(osbuild.NewWAAgentConfStage(waConfig))
+	if keyboard != nil {
+		pl.Keyboard = keyboard
+	} else if imageConfig.Keyboard != nil {
+		pl.Keyboard = &imageConfig.Keyboard.Keymap
 	}
 
-	if pt != nil {
-		kernelOptions := osbuild.GenImageKernelOptions(pt)
-		if t.kernelOptions != "" {
-			kernelOptions = append(kernelOptions, t.kernelOptions)
-		}
-		if bpKernel := c.GetKernel(); bpKernel.Append != "" {
-			kernelOptions = append(kernelOptions, bpKernel.Append)
-		}
-		p = prependKernelCmdlineStage(p, strings.Join(kernelOptions, " "), pt)
-		p.AddStage(osbuild.NewFSTabStage(osbuild.NewFSTabStageOptions(pt)))
-		kernelVer := rpmmd.GetVerStrFromPackageSpecListPanic(packages, c.GetKernel().Name)
-		bootloader := bootloaderConfigStage(t, *pt, kernelVer, false, false)
-
-		if cfg := imageConfig.Grub2Config; cfg != nil {
-			if grub2, ok := bootloader.Options.(*osbuild.GRUB2StageOptions); ok {
-				// grub2.Config.Default is owned and set by `NewGrub2StageOptionsUnified`
-				// and thus we need to preserve it
-				if grub2.Config != nil {
-					cfg.Default = grub2.Config.Default
-				}
-			}
-		}
-
-		p.AddStage(bootloader)
+	if hostname := c.GetHostname(); hostname != nil {
+		pl.Hostname = *hostname
 	}
 
-	p.AddStage(osbuild.NewSELinuxStage(selinuxStageOptions(false)))
-
-	if t.rpmOstree {
-		p.AddStage(osbuild.NewOSTreePrepTreeStage(&osbuild.OSTreePrepTreeStageOptions{
-			EtcGroupMembers: []string{
-				// NOTE: We may want to make this configurable.
-				"wheel", "docker",
-			},
-		}))
+	timezone, ntpServers := c.GetTimezoneSettings()
+	if timezone != nil {
+		pl.Timezone = *timezone
+	} else {
+		pl.Timezone = imageConfig.Timezone
 	}
 
-	return p, nil
+	if len(ntpServers) > 0 {
+		pl.NTPServers = ntpServers
+	} else if imageConfig.TimeSynchronization != nil {
+		pl.NTPServers = imageConfig.TimeSynchronization.Timeservers
+	}
+
+	pl.Grub2Config = imageConfig.Grub2Config
+	pl.Sysconfig = imageConfig.Sysconfig
+	pl.SystemdLogind = imageConfig.SystemdLogind
+	pl.CloudInit = imageConfig.CloudInit
+	pl.Modprobe = imageConfig.Modprobe
+	pl.DracutConf = imageConfig.DracutConf
+	pl.SystemdUnit = imageConfig.SystemdUnit
+	pl.Authselect = imageConfig.Authselect
+	pl.SELinuxConfig = imageConfig.SELinuxConfig
+	pl.Tuned = imageConfig.Tuned
+	pl.Tmpfilesd = imageConfig.Tmpfilesd
+	pl.PamLimitsConf = imageConfig.PamLimitsConf
+	pl.Sysctld = imageConfig.Sysctld
+	pl.DNFConfig = imageConfig.DNFConfig
+	pl.SshdConfig = imageConfig.SshdConfig
+	pl.AuthConfig = imageConfig.Authconfig
+	pl.PwQuality = imageConfig.PwQuality
+	pl.WAAgentConfig = imageConfig.WAAgentConfig
```

Comment:
I didn't want to get into the inheritance stuff, so will leave this all stupid/open-coded for now. I thin it will be clearer what we need once we have cleaned things up a bit further.